frontend/src/components/ScheduleCreation.vue

```diff
@@ -673,18 +691,73 @@ watch(
           </label>
         </div>
       </div>
-      <!-- option to deactivate confirmation -->
-      <div class="px-4">
-        <switch-toggle
-          class="my-1 pr-3 text-sm font-medium text-gray-500 dark:text-gray-300"
-          :active="schedule.booking_confirmation"
-          :label="t('label.bookingConfirmation')"
-          :disabled="!scheduleInput.active"
-          @changed="toggleBookingConfirmation"
-          no-legend
-        />
-        <div class="text-xs">
-          {{ t('text.ownerNeedsToConfirmBooking') }}
+
+      <!-- step 4 -->
+      <div
+        @click="state = ScheduleCreationState.Booking"
+        class="btn-step-3 mx-4 flex flex-col gap-2 rounded-lg border border-zinc-200 p-4 text-gray-700 dark:border-gray-500 dark:bg-gray-600 dark:text-gray-100"
+        :class="{'bg-neutral-50': activeStep4}"
+        id="schedule-details"
+      >
+        <div class="flex cursor-pointer items-center justify-between">
+          <div class="flex flex-col gap-1">
+            <h2>
+              {{ t("label.bookingSettings") }}
+            </h2>
+            <h3 class="text-xs">
+              {{ t('label.scheduleSettings.bookingSettingsSubHeading') }}
+            </h3>
+          </div>
+          <icon-chevron-down
+            class="size-6 -rotate-90 fill-transparent stroke-gray-800 stroke-1 transition-transform dark:stroke-gray-100"
+            :class="{ '!rotate-0': activeStep4 }"
+          />
+        </div>
+        <div v-show="activeStep4" class="flex flex-col gap-2">
+          <hr/>
+          <!-- custom quick link -->
+          <label class="relative flex flex-col gap-1">
+            <div class="font-medium text-gray-500 dark:text-gray-300">
+              {{ t("label.quickLink") }}
+            </div>
+            <div class="flex gap-2">
+              <text-input
+                type="text"
+                name="slug"
+                :prefix="`/${user.data.username}/`"
+                v-model="slugInput"
+                class="w-full rounded-md disabled:cursor-not-allowed"
+              />
+              <refresh-icon class="cursor-pointer mt-2.5 text-teal-600" @click.prevent="refreshSlug" />
+            </div>
+          </label>
+          <!-- option to deactivate confirmation -->
+          <div class="flex flex-col gap-3">
+            <switch-toggle
+              class="my-1 text-sm font-medium text-gray-500 dark:text-gray-300"
+              :active="schedule.booking_confirmation"
+              :label="t('label.bookingConfirmation')"
+              :disabled="!scheduleInput.active"
+              @changed="toggleBookingConfirmation"
+              no-legend
+            />
+            <div class="whitespace-pre-line rounded-lg bg-white p-4 text-xs text-gray-500 dark:bg-gray-800">
+              <div>
+                {{ t('text.yourQuickLinkIs', { url: user.myLink }) }}<br />
+                <i18n-t keypath="text.toUpdateYourUsername.text" for="text.toUpdateYourUsername.link">
```

Comment:
I reccommend using the slot-based method which I discovered recently
https://vue-i18n.intlify.dev/guide/advanced/component.html#slots-syntax-usage

It's slightly more readable.

Reply:
Ah nice! Changed it.